src/predicates/intersects.jl

```diff
@@ -33,6 +33,16 @@ intersects(r::Ray, t::Triangle) = !isnothing(r ∩ t)
 
 intersects(t::Triangle, r::Ray) = intersects(r, t)
 
+function intersects(ray::Ray{Dim,T}, sphere::Union{Ball{Dim,T},Sphere{Dim,T}})::Bool where {Dim,T}
```

Comment:
Let's write a method for `Sphere` and then add a separate method for `Ball` that simply calls `boundary(ball)` and forwards the sphere to the existing method.

Please rename the arguments of the function to single letters as in the other methods. Try to follow the code style of the surrounding code as much as possible: no underscores in variable names, no `return` at the last line, etc.